docs/enhancements/2023-09-01-cross-silo-error.md

```diff
@@ -0,0 +1,28 @@
+## Summary
+### General Motivation
+Before this proposal, when the execution of a DAG encounters an error in 'alice', below is what will happen:
+![image](https://github.com/ray-project/rayfed/assets/26196566/77fa1bd1-dc14-4ddc-8d9e-5dbaf6942564)
+
+In alice, both main thread and data sending thread will raise the error, and the process will exit.
+In bob,  since it needs the input from 'alice', it will waits for 'alice' forever no matter whether 'alice' exists or not.
+
+Therefore, we need a mechanism to inform the other participant when the DAG execution raises error.
+
+## Design and Architecture
+The below graph shows what will happen now after this proposal:
+![image](https://github.com/ray-project/rayfed/assets/26196566/2f2ad25d-53cb-4ba8-9f3b-38419c7f494b)
+
+In alice, when the data-sending thread finds a RayTaskError indicating a execution failure, it will wrap it to a `FedRemoteError` object and replace the original data object in place to send to bob.
```

Comment:
```suggestion
In alice, when the data-sending thread finds a RayTaskError indicating an execution failure, it will wrap it as a `FedRemoteError` object and replace the original data object in place to send to bob.
```